week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,11 +1,91 @@
-function setAlarm() {}
+let backgroundColors;
+let intervalId = -1;
+let inputValue = 0;
+let remainingTime = 0;
 
+function changeBackgroundColor() {
+  backgroundColors = setInterval(changeColor, 500);
+}
+function changeColor() {
+  let body = document.querySelector("body");
+  body.style.backgroundColor =
+    body.style.backgroundColor == "crimson" ? "#0083B0" : "crimson";
+}
+function stopChangeBackgroundColor() {
+  clearInterval(backgroundColors);
+}
+
+function fancyTimeFormat(time) {
+  let minutes = Math.floor(time / 60);
+  let seconds = time - minutes * 60;
+  let newTime = "";
+  newTime +=
+    "" + minutes.toString().padStart(2, "0") + ":" + (seconds < 10 ? "0" : "");
+  newTime += "" + seconds;
+  return newTime;
+}
+
+function setAlarm() {
+  clearInterval(intervalId);
+  let timeRemain = document.getElementById("timeRemaining");
+  inputValue = document.getElementById("alarmSet").value;
+  if (inputValue == "") {
+    setAlarm();
+  }
+  timeRemain.textContent = `Time Remaining: ${fancyTimeFormat(inputValue)}`;
+
+  intervalId = setInterval(function () {
+    inputValue--;
+    timeRemain.textContent = `Time Remaining: ${fancyTimeFormat(inputValue)}`;
+    if (inputValue == 0) {
```

Comment:
I can put Number(inputValue) === 0

Reply:
That definitely solves the problem of using `==` but having to wrap `inputValue` with `Number()` every time it's being used is another code smell 😅 Ideally there should be a single place where that happens (the general principle is called [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)).

How would you change the code so that `Number()` is used just once? To clarify, I'm not suggesting to create a function here (it's not necessary).